lib/syskit/roby_app/plugin.rb

```diff
@@ -146,6 +150,58 @@ def self.setup(app)
 
                 rtt_core_model = app.default_loader.task_model_from_name("RTT::TaskContext")
                 Syskit::TaskContext.define_from_orogen(rtt_core_model, register: true)
+
+                # Log Transfer FTP Server spawned during Application#setup
+                app.setup_local_log_transfer_server if app.log_transfer_ip
+            end
+
+            def setup_local_log_transfer_server
+                @log_transfer_user = "process server"
+                @log_transfer_password = SecureRandom.base64(15)
+
+                tmp_root_ca = TmpRootCA.new(log_transfer_ip)
+                @log_transfer_certificate = tmp_root_ca.certificate
+
+                start_local_log_transfer_server(log_dir, @log_transfer_user, @log_transfer_password, tmp_root_ca.private_certificate_path)
+
+                @log_transfer_port = @log_transfer_server.port
+            end
+
+            def send_file_transfer_command(name, logfile)
+                # Establishes communication with said process server
+                client = Syskit.conf.process_server_for(name)
+
+                # Commands method log_upload_file from said process server
+                client.log_upload_file(
+                    log_transfer_ip,
+                    @log_transfer_port,
+                    @log_transfer_certificate,
+                    @log_transfer_user,
+                    @log_transfer_password,
+                    logfile
+                )
+                client
+            end
+
+            def start_local_log_transfer_server(tgt_dir, user, password, private_certificate_path)
+                if @log_transfer_server
+                    raise "log transfer server is already started"
+                end
+
+                @log_transfer_server = Syskit::RobyApp::LogTransferServer::SpawnServer.new(
+                    tgt_dir,
+                    user,
+                    password,
+                    private_certificate_path
+                )
+            end
+
+            def stop_local_log_transfer_server
+                if @log_transfer_server
+                    @log_transfer_server&.stop
```

Comment:
'&.' means "check if the receiver is nil, and return nil if it is instead of calling the method"

Since you check first that @log_transfer_server is not nil, '&.' is meaningless here.